tasks/us/hierarchy.py

```diff
@@ -0,0 +1,332 @@
+        GROUP BY child_id, child_level
+        HAVING count(1) > 1
+    """
+
+    def requires(self):
+        return {
+            'level': USLevelHierarchy(year=self.year,
+                                      current_geography=self.current_geography,
+                                      parent_geography=self.parent_geography),
+            'current_geom': ShorelineClip(year=self.year,
+                                          geography=self.current_geography),
+            'parent_geom': ShorelineClip(year=self.year,
+                                         geography=self.parent_geography)
+        }
+
+    def run(self):
+        session = current_session()
+        sql = '''
+            UPDATE {table}
+            SET weight = ST_Area(
+                ST_Intersection(cgt.the_geom, pgt.the_geom), False)
+            FROM
+                observatory.{current_geom_table} cgt,
+                observatory.{parent_geom_table} pgt
+            WHERE cgt.geoid = {table}.child_id
+              AND pgt.geoid = {table}.parent_id
+              AND (child_id, child_level) IN (
+              {unweighted_child_sql}
+        )
+        '''
+        table = self.input()['level'].qualified_tablename
+        session.execute(
+            sql.format(
+                table=table,
+                current_geom_table=self.input()['current_geom'].get(
+                    session).tablename,
+                parent_geom_table=self.input()['parent_geom'].get(
+                    session).tablename,
+                unweighted_child_sql=self.UNWEIGHTED_CHILD_SQL.format(
+                    table=table)
+            )
+        )
+        session.commit()
+
+    def complete(self):
+        return len(current_session().execute(self.UNWEIGHTED_CHILD_SQL.format(
+            table=self.input()['level'].qualified_tablename)).fetchall()) == 0
+
+    def output(self):
+        return self.input()['level']
+
+
+
+class USLevelHierarchy(TempTableTask):
+    year = IntParameter()
+    current_geography = Parameter()
+    parent_geography = Parameter()
+
+    def requires(self):
+        return {
+            'current_info': USLevelInfo(year=self.year,
+                                        geography=self.current_geography),
+            'current_geom': ShorelineClip(year=self.year,
+                                          geography=self.current_geography),
+            'parent_info': USLevelInfo(year=self.year,
+                                       geography=self.parent_geography),
+            'parent_geom': ShorelineClip(year=self.year,
+                                         geography=self.parent_geography)
+        }
+
+    def target_tablename(self):
+        return abbr_tablename(super(USLevelHierarchy, self).target_tablename(),
+                              [self.current_geography, self.parent_geography],
+                              self.year)
+
+    def run(self):
+        session = current_session()
+        input = self.input()
```

Comment:
You are shadowing `input`, a built-in function :sweat_smile:

Reply:
Is there a "pythonic" convention about this? I need a short identifier (len <= 7) and I don't have any good candidate. I'll use `_input` if there's no better proposal.

Reply:
It would be better to use `input_` (trailing underscore is normally used for avoiding conflict with Python keywords and leading underscore is normally used for declaring private variables)